### ISSUE TITLE
Set VAULT_ADDR in tasks when VAULT_TOKEN is set

### ISSUE THREAD

**Issue by blalor** [hashicorp/nomad#2393]
Nomad v0.5.4

When Nomad schedules a task that needs a Vault token, that task gets a `VAULT_TOKEN` environment variable.  It does *not* get a `VAULT_ADDR` var set, however.  It would be nice if operators didn't have to set this explicitly in the jobspec.

**Comment by dadgar** [comment]
Hey @blalor,

I agree it would be nice. The reason we have not done this is that the address Nomad may be connecting to Vault with may not be routable from the network the task is in, especially as there are network plugins.

So unfortunately I am going to close this. Please let me know if my explanation didn't make sense.

I do suggest just routing to vault.service.consul.

**Comment by blalor** [comment]
How about providing it as a default, with a caveat that it may need to be overridden?

**Comment by dadgar** [comment]
I think maybe in the future we will let operators inject default environment variables so it can be operator configured. I would not like Nomad to inject environment variables that will ever be wrong.

**Comment by aantono** [comment]
I was just about to submit a PR for this feature, but good thing I've stumbled on this ticket first.  I think it definitely should be considered to somehow provide an "automated" way of exposing `VAULT_ADDR` env along with `VAULT_TOKEN` if `vault` stanza is used in job config.  Otherwise it kinda violates the transparent config premise, sine the user submitting the Job needs to not only know that it needs a vault token, but also needs to actually explicitly know where Vault is running *AND* make sure that it is the *SAME* Vault cluster that Nomad servers use to procure the tokens from.  That seems like a significant coupling unnecessarily pushing concerns outside the cluster perimeter. 

While I agree with your explanation premise that potentially there could be network firewalls and other hurdles, in reality that's mostly not the case, and having a configuration option to tell Nomad Clients to export `VAULT_ADDR` using the configured address - https://www.nomadproject.io/docs/agent/configuration/vault.html#address  Also, come to think of it, if every Nomad Client agent requires `vault { address= }` stanza to be present in the agent config (and I've verified that by removing it and getting an error on Vault Client agent restart), I don't see how it will be unreachable by the task job running on that same client node, since the Vault Token that the Client agent is unwrapping is being performed against that configured Vault, so it would have to be reachable (Unless I'm missing something in my understanding of the architecture).
